src/dialogs/symlink.tsx

```diff
@@ -0,0 +1,254 @@
+ */
+import React from 'react';
+
+import { Button } from '@patternfly/react-core/dist/esm/components/Button';
+import { Form, FormGroup } from "@patternfly/react-core/dist/esm/components/Form";
+import { Modal, ModalVariant } from '@patternfly/react-core/dist/esm/components/Modal';
+import { Popover } from "@patternfly/react-core/dist/esm/components/Popover";
+import { Radio } from "@patternfly/react-core/dist/esm/components/Radio";
+import { TextInput } from "@patternfly/react-core/dist/esm/components/TextInput";
+import { HelpIcon } from "@patternfly/react-icons";
+
+import cockpit, { BasicError } from 'cockpit';
+import { fsinfo, type FileInfo } from 'cockpit/fsinfo.ts';
+import { FormHelper } from 'cockpit-components-form-helper';
+import { InlineNotification } from 'cockpit-components-inline-notification';
+import { dirname } from 'cockpit-path.ts';
+import type { Dialogs, DialogResult } from 'dialogs';
+import { useInit } from 'hooks.ts';
+import { superuser } from 'superuser';
+import { fmt_to_fragments } from "utils";
+
+import { useFilesContext, type FolderFileInfo } from '../common.ts';
+import { get_owner_candidates } from '../ownership.tsx';
+
+function checkLinkName(candidate: string, entries: Record<string, FileInfo>, selectedFile?: FolderFileInfo) {
+    if (candidate === "") {
+        return _("Name cannot be empty");
+    } else if (candidate.length >= 256) {
+        return _("Name too long");
+    } else if (selectedFile && selectedFile.name === candidate) {
+        return _("Filename is the same as original name");
+    } else if (candidate in entries) {
+        if (entries[candidate].type === "dir") {
+            return _("Directory with the same name exists");
+        }
+        return _("File exists");
+    } else {
+        return null;
+    }
+}
+
+const _ = cockpit.gettext;
+
+type symlink_mode = "absolute" | "relative";
+
+const CreateLinkModal = ({ dialogResult, path, selected } : {
+    dialogResult: DialogResult<void>
+    path: string,
+    selected: FolderFileInfo,
+}) => {
+    const [errorMessage, setErrorMessage] = React.useState<string | undefined>();
+    const [symlinkName, setSymlinkName] = React.useState<string>(cockpit.format(_("link to $0"), selected.name));
+    const [symlinkError, setSymlinkError] = React.useState<string | null>(null);
+    const [mode, setMode] = React.useState<symlink_mode>("absolute");
+    const [user, setUser] = React.useState<cockpit.UserInfo| undefined>();
+
+    const { cwdInfo } = useFilesContext();
+
+    useInit(() => {
+        cockpit.user().then(user => setUser(user));
+    });
+
+    const createLink = async () => {
+        cockpit.assert(user !== undefined);
+        const cmd = ["ln", "--symbolic"];
+        const target = `${path}${selected.name}`;
+        const is_dir = symlinkName.startsWith('/') || symlinkName.startsWith('..');
+        let owner = null;
+
+        if (superuser.allowed && user && cwdInfo) {
```

Comment:
This is another bug, fsinfo only works when `superuser.allowed` but it should not.